pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -321,11 +308,13 @@ func (r *ReconcileClusterPool) Reconcile(ctx context.Context, request reconcile.
 
 	availableCapacity := math.MaxInt32
 	if clp.Spec.MaxSize != nil {
-		availableCapacity = int(*clp.Spec.MaxSize) - len(unClaimedCDs) - len(claimedCDs)
+		availableCapacity = int(*clp.Spec.MaxSize) - cds.Total()
+		numUnassigned := len(cds.Installing()) + len(cds.Assignable())
+		numAssigned := cds.Total() - numUnassigned
```

Comment:
doesn't total include the deleting+markedForDeletion ? those should not be included in numAssigned ?